wsagent/che-core-api-languageserver/src/main/java/org/eclipse/che/api/languageserver/registry/ServerInitializerImpl.java

```diff
@@ -142,9 +142,16 @@ protected void registerCallbacks(LanguageServer server, LanguageServerLauncher l
     }
   }
 
-  private InitializeParams prepareInitializeParams(String projectPath) {
+  private InitializeParams prepareInitializeParams(
+      LanguageServerLauncher launcher, String projectPath) {
```

Comment:
Looking at that code I think that you've struggled to fit a new feature in a non-conforming architecture of components. Have you thought about refactoring of this subsystem to more agnostic to local LS launching?

Reply:
You are absolutely right, there was some struggle, but tell me this, how on earth did you guess that I sometimes really think of making this subsystem more agnostic to local LS launching? Unfortunately it is not in the nearest plans yet.